contracts/Karmic.sol

```diff
@@ -6,19 +6,57 @@ import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "./Badger.sol";
 
 contract Karmic is Badger{
-    mapping(address => uint256) private boxTokenTiers;
+    mapping(address => BoxToken) private boxTokenTiers;
     uint256 boxTokenCounter;
+    uint256 threshold;
 
-    constructor(string memory _newBaseUri) Badger(_newBaseUri) {}
+    struct BoxToken{
+        uint256 id;
+        uint256 amount;
+        uint256 funds;
+        bool passedThreshold;
+    }
+
+    modifier isBoxToken(address token){
+        require(boxTokenTiers[token].id != 0, "It is not a box token");
+        _;
+    }
+
+
+    constructor(string memory _newBaseUri, uint256 _threshold) Badger(_newBaseUri) {
+        boxTokenCounter = 1;
+        threshold = _threshold;
+        createTokenTier(0, _newBaseUri, false, address(0));
+    }
+
+    fallback() external payable{
+        if(boxTokenTiers[msg.sender].id != 0){
+            boxTokenTiers[msg.sender].amount = ERC20(msg.sender).totalSupply();
+            boxTokenTiers[msg.sender].funds = msg.value;
+            boxTokenTiers[msg.sender].passedThreshold = msg.value >= threshold;
```

Comment:
Seems like this is the only time that we're using threshold. Can we instead of storing threshold on the constructor and checking against it, call the Mirro contracts and ask for the threshold per token? It's A) cheaper, and B) allows for using different fundingGoals for potential future crowdfunding campaigns on mirror?

Reply:
Hmm, I like this idea generally, but after looking into the mirror crowdfund [contracts](https://github.com/mirror-xyz/crowdfund/blob/main/contracts/) I am not sure if it is possible.

Reply:
There is a value, called `fundingCap`, but it is clearly not the threshold, because it is impossible to fund more then the `fundingCap` due to the code of payable function in the CrowdfundLogic, but if we look into the mirror [itself](https://mint.mirror.xyz/mDws2xGmJCq55ehGDdcv8tGDCpSd_c_CfeqhexiKSqU) it's definitely possible to crowdfund more then the threshold....

Reply:
Maybe we should just have a specific threshold for each boxTokenTier, wdyt?

Reply:
Thresholds are now deeply personal